zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/archiver/ElasticsearchRepositoryIT.java

```diff
@@ -203,7 +204,75 @@ void shouldReindexDocuments() throws IOException {
         .containsExactlyInAnyOrder("1", "2", "3");
   }
 
-  private void index(final String index, final TestDocument document) {
+  @Test
+  void shouldGetProcessInstancesNextBatch() throws IOException {
+    // given - 5 documents, two of which were created over an hour ago, one of which was created,
```

Comment:
4 documents right?
❓ Also I am not sure if I understood the test, are we not supposed to get the documents 1 and 2 here? since both are process instances and were created two hours ago? But I checked locally and its passing

Reply:
Yeah, 4, initially I had planned 5 but realized it's not required.

So we have 4 docs:

```java
new TestProcessInstance(
  "1", twoHoursAgo, ListViewTemplate.PROCESS_INSTANCE_JOIN_RELATION, 1);
new TestProcessInstance(
  "2", twoHoursAgo, ListViewTemplate.PROCESS_INSTANCE_JOIN_RELATION, 2);
new TestProcessInstance("3", twoHoursAgo, ListViewTemplate.ACTIVITIES_JOIN_RELATION, 1);
new TestProcessInstance(
  "4", now.toString(), ListViewTemplate.PROCESS_INSTANCE_JOIN_RELATION, 1);
```

We're archiving _only for partition 1_. So we only have "1", "3", and "4" for partition 1. Then we only select those from the `PROCESS_INSTANCE_JOIN_RELATION` - so that's only "1", and "4". Finally, "4" was created right now, so we only expect "1" which was created over 1h ago (our archiving time point).

Reply:
ah right the partitions here make difference! Thanks 🙌